elasticdl/python/master/master.py

```diff
@@ -52,19 +52,30 @@ def run(self):
         Dispatch the tasks to the workers until all the tasks are completed.
         """
         try:
-            while True:
-                if self.task_manager.finished():
-                    if self.pod_manager:
-                        self.pod_manager.update_status(
-                            InstanceManagerStatus.FINISHED
-                        )
-                    break
-                if self.pod_manager.all_workers_exited:
-                    raise Exception(
-                        "All workers exited but there also are",
-                        "unfinished tasks",
-                    )
-                time.sleep(30)
+            no_loop_run = not self.task_manager and not self.pod_manager
```

Comment:
I'm wondering whether the scenario exists that neither task_manager nor pod_manager is created.

Reply:
When users set arguments wrong.

Reply:
Maybe can do some argument check and raise ValueError directly.

Reply:
Maybe we should assume that when it `run`, it should satisfy that at least one manager is running. 
Then we need to add an argument validation in `init` or `prepare`.